crates/service/src/adapters.rs

```diff
@@ -284,12 +284,121 @@ impl<T> FromIterator<T> for VecInput<T> {
     }
 }
 
+/// Adapter for using a [`ServiceMonitor`] as an input source for listener services.
+///
+/// This adapter watches for status updates from another service and yields them as
+/// messages. The listener is passive and unaware to the monitored service.
+///
+/// When the watch channel closes (monitored service exits), this adapter returns
+/// `None` to signal the listener should also exit.
+pub struct StatusMonitorInput<S: Service> {
+    rx: watch::Receiver<S::Status>,
+    closed: bool,
+    /// Optional Tokio runtime handle for sync services.
+    /// Required when using this input with `SyncService`.
+    handle: Option<tokio::runtime::Handle>,
+}
```

Comment:
This only being optional makes is so there's nothing stopping you from misusing it.  We can just not implement the sync trait on this and rely on a consumer to use the adapter if they want to use it from a sync context.

Also in line with the other points, we could loosen this up a bit and call it `TokioWatchInput` and make it generic over just the message type instead of the whole service.